src/misc.js

```diff
@@ -265,4 +265,19 @@ engine.singleton = function (coll, type) {
   throw new Error('Not supported type ' + type);
 };
 
+/**
+ * Checks whether a primitve value is present
+ */
+engine.hasValueFn = function(coll) {
+  if (coll.length === 1){
+    return [isPrimitive(coll[0].data)];
```

Comment:
We are currently using `util.valData(coll[0])` instead of `coll[0].data`, but I guess it is not a big deal in this case.
Maybe I'm misunderstanding the spec, but it says "Otherwise, the return value is empty."
See http://hl7.org/fhir/fhirpath.html#functions
In JS implementation, empty is an empty array.

Reply:
Concerning the specification, I concur with you regarding the empty array. Strangely the [simplifier implementation](https://simplifier.net/fhirpath?filepath=package/CapabilityStatement-base.json&scope=hl7.fhir.r4.core@4.0.1&query=) and the [pascal reference implementation](https://github.com/HealthIntersections/fhirserver/blob/master/library/fhir4/fhir4_pathengine.pas) both return false... which made me stumble a bit... But yeah, I guess according to the implementation an empty array should be returned.

Reply:
Grahame says it can return false, so there may be three possible return values.  https://chat.fhir.org/#narrow/stream/179266-fhirpath/topic/hasValue